src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -181,16 +191,28 @@ private function generateItems(array $keys, $now)
     {
         $f = $this->createCacheItem;
 
-        foreach ($keys as $key) {
-            if (!$isHit = isset($this->expiries[$key]) && ($this->expiries[$key] >= $now || !$this->deleteItem($key))) {
-                $value = null;
-            } elseif ($this->storeSerialized) {
-                $value = unserialize($this->values[$key]);
-            } else {
-                $value = $this->values[$key];
+        try {
+            foreach ($keys as $i => $key) {
+                if (!$isHit = isset($this->expiries[$key]) && ($this->expiries[$key] >= $now || !$this->deleteItem($key))) {
+                    $value = null;
+                } elseif (!$this->storeSerialized) {
+                    $value = $this->values[$key];
+                } elseif ('b:0;' === $value = $this->values[$key]) {
+                    $value = false;
+                } elseif (false === $value = unserialize($value)) {
+                    $value = null;
+                    $isHit = false;
+                }
+                unset($keys[$i]);
+
+                yield $key => $f($key, $value, $isHit);
             }
+        } catch (\Exception $e) {
```

Comment:
Shouldn't this be moved inside the foreach to fetch as many values as possible ?